examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,338 @@
+  }
+
+  /**
+   *
+   * @param onCloudLoginSuccess
+   */
+  async login(onCloudLoginSuccess) {
+    const link = `${window.location.protocol}//${window.location.host}/${AWS_LOGIN_URL}`;
+    const style = `location, toolbar, resizable, scrollbars, status, width=500, height=440, top=200, left=400`;
+    const authWindow = window.open(link, 'awsCognito', style);
+
+    // if authWindow is null, it could not be opened
+    const handleLogin = async e => {
+      if (authWindow.location.href === link) {
+        if (authWindow) {
+          authWindow.close();
+        }
+
+        window.removeEventListener('message', handleLogin);
+
+        if (e.data.success) {
+          this._getUserInfo()
+            .then(currentUser => {
+              this._currentUser = currentUser;
+            })
+            .catch();
+          onCloudLoginSuccess();
+        }
+      }
+    };
+    window.addEventListener('message', handleLogin);
+  }
+
+  /**
+   *
+   * @returns {Array<Viz>}
+   */
+  async listMaps() {
+    const publicMaps = this._getMapListFromStorage('public');
+    const privateMaps = this._getMapListFromStorage('private');
+    return Promise.all([publicMaps, privateMaps])
+      .then(values => values.flat())
+      .catch(e => AwsProvider._handleError(e));
+  }
+
+  _getMapListFromStorage(level) {
+    return Storage.list('', {level})
+      .then(result => AwsProvider._prepareFileList(result, level))
+      .catch(e => {
+        const capitalizedLevel = level.charAt(0).toUpperCase() + level.slice(1);
+        AwsProvider._handleError(`${capitalizedLevel} maps failed to load`, e);
+      });
+  }
+
+  /**
+   * Generates array of viz objects from list of files from storage
+   * @returns {Array<Viz>}
+   */
+  static async _prepareFileList(fileList, level) {
+    const mapFileList = fileList.filter(file => file.key.endsWith('map.json'));
+
+    const updatedFileList = mapFileList.map(file => {
+      const title = file.key.split('.')[0];
+      const thumbnailKey = `${title}.thumbnail.png`;
+      const metaKey = `${title}.meta.json`;
+
+      const fileData = {
+        id: file.key,
+        title,
+        description: '',
+        privateMap: level === 'private',
+        thumbnail: '',
+        lastModification: new Date(Date.parse(file.lastModified)),
+        loadParams: {
+          identityId: '',
+          mapId: file.key,
+          level
+        }
+      };
+
+      const thumbnailUrl = fileList.some(f => f.key === thumbnailKey)
+        ? this._getFile(thumbnailKey, 'image', {level, download: false})
+        : null;
+      const description = fileList.some(f => f.key === metaKey)
+        ? this._getFile(metaKey, 'meta data', {level, download: true})
+        : 'No description meta data file';
+
+      return Promise.all([thumbnailUrl, description])
+        .then(respList => {
+          fileData.thumbnail = respList[0];
+          fileData.description = respList[1];
+        })
+        .then(resp => fileData)
```

Comment:
Maybe: You can also skip creating the `fileData` structure, sth. like:
```
const updatedFileList = mapFileList.map(file => {
      const title = file.key.split('.')[0]
      const thumbnailKey = `${title}.thumbnail.png`
      const metaKey = `${title}.meta.json`

      const loadThumbnail = fileList.some(f => f.key === thumbnailKey)
        ? this._getFile(thumbnailKey, 'image', {level, download: false})
        : null
      const loadDescription = fileList.some(f => f.key === metaKey)
        ? this._getFile(metaKey, 'meta data', {level, download: true})
        : 'No description meta data file'

      return Promise.all([loadThumbnail, loadDescription])
        .then(([thumbnail, description]) => ({
          id: file.key,
          title,
          description: '',
          privateMap: level === 'private',
          thumbnail: '',
          lastModification: new Date(Date.parse(file.lastModified)),
          loadParams: {
            identityId: '',
            mapId: file.key,
            level
          },
          thumbnail,
          description
        }))
        .catch(e => this._handleError(e))
    })
```

Reply:
Very nice, thnx :)